### DISCUSSION TITLE
Idiomatic way to call Lua function?

### DISCUSSION THREAD

**Discussion by JLPLabs** [pallene-lang/pallene#484] [Q&A] [answered]
From the documentation (emphasis added)..."Pallene functions can be called from Lua and _**Pallene can call Lua functions**_ as well."

I would like to call a Lua function (for instance, `io.write()` -- which I know is not built into Pallene, but perhaps I can create a simple library that contains it); however, since `require` is not part of Pallene, I do not know how to make my little library of Lua functions available to Pallene...

Is there an idiomatic way to call Lua functions from Pallene? 

Many thanks!

**Comment by hugomg** [comment, marked as answer]
Currently this can be a bit difficult because we don't have require. We're working on implementing require, but it's not ready yet.

The workaround that is possible for now is that you can have Lua call a Pallene function, passing a Lua function as an argument. Or even passing a table of functions.

**Reply by JLPLabs** [reply]
Thank you for the quick response! I will try it. If I get something that is reasonable I may post it back here as an example for my future self...    :-)
